### DISCUSSION TITLE
Theme support for styling purposes

### DISCUSSION THREAD

**Discussion by spd-spd** [RobertWeinmeister/dokuwiki-mermaid#16] [Q&A] [answered]
One of its many issues working with link styling: https://github.com/mermaid-js/mermaid/issues/3121

One of the workaround answers requires this in the config in the online editor:

{
  "theme": "neutral"
}

Are we able to support this in our favorite (dokuwiki-mermaid) plugin?  If it is it already, I just need the documentation.

**Comment by RobertWeinmeister** [comment, marked as answer]
```%%{init: {'theme':'neutral'}}%%``` right at the beginning should do the trick. Have a look at [diagram specific themes](https://mermaid.js.org/config/theming.html#diagram-specific-themes).

**Comment by spd-spd** [comment]
Thanks, I see there are other examples of that syntax on the doc page, but not quite that one:
%%{init: {"flowchart": {"htmlLabels": false}} }%%
%%{ init: { 'flowchart': { 'curve': 'stepBefore' } } }%%
%%{init: {"flowchart": {"defaultRenderer": "elk"}} }%%

**Reply by RobertWeinmeister** [reply]
Each diagram has its own possible [styling and configuration](https://mermaid.js.org/syntax/flowchart.html#styling-and-classes). They all work with this plugin.

**Comment by RobertWeinmeister** [comment]
The five default themes are available via the [plugin configuration](https://github.com/RobertWeinmeister/dokuwiki-mermaid?tab=readme-ov-file#default-theme).
